### PR TITLE
Use users $HOME/.ssh for openstack_tmp SSH key

### PR DESCRIPTION
**Pull request by chaosaffe**
**What this PR does / why we need it**:
Uses the users `$HOME` directory to store and reference the temporary SSH key generated to access the master nodes. This eliminates the need for MacOS users to create a fake `/root`, as well as enabling Linux users to run commands without root privileges.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
N/A

**Special notes for your reviewer**:
N/A

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cloud/openstack/deployer.go [hunk 1]
```diff
@@ -56,8 +57,13 @@ func (d *DeploymentClient) GetKubeConfig(cluster *clusterv1.Cluster, master *clu
 		return "", err
 	}
 
+	homeDir, ok := os.LookupEnv("HOME")
```
Comment (flaper87) [inline cloud/openstack/deployer.go]: Should we try to read `MACHINE_CONTROLLER_SSH_HOME` before building the path ourselves?

Comment (chaosaffe) [reply cloud/openstack/deployer.go]: `MACHINE_CONTROLLER_SSH_HOME` is only defined in `generate-yaml.sh` and is not exported. This is just intermediary work to allow MacOS users to use the provider to bootstrap without a lot of overhead (and remove the dependence on running as `root` on Linux). Once #36 lands I intend to add a patch to enable the user to configure the directory to use via a flag. 
I know it is fairly common in the OpenStack world to use env vars, but I would really like to see us avoid using implicit inputs (i.e. env vars) and opt for explicit inputs (i.e. flags, config files)

Comment (flaper87) [reply cloud/openstack/deployer.go]: wfm! thanks

**Pull request merged**